### ISSUE TITLE
add ECDH support

### ISSUE THREAD

**Issue by Dvergatal** [tpm2-software/tpm2-pkcs11#661]
Hi @williamcroberts again,
i understand that you have closed #656, but you haven't answered me for my question about adding support for ECDH.

**Comment by williamcroberts** [comment]
We can add it, once we know how to do it in the tpm2-tools project. It's a buit of an ongoing open issue. @idesai has made some progress on it, I think he has something working as far as TPM output, but we want to make sure it works end-to-end with an OSSL peer.

**Comment by Dvergatal** [comment]
Ok thx. I haven't noticed that comment yesterday:P

**Comment by williamcroberts** [comment]
FYI it was implemented in the openssl engine project as:
  - https://github.com/tpm2-software/tpm2-tss-engine/pull/207

I'm still not sure what it looks like using tpm2-pkcs11 API to generate an ECDH key and using it, as far as flow yet. But Ill start looking into it.

**Comment by Dvergatal** [comment]
Hmmmm there is only EC key. ECDH is an algorithm used i.e. in TLS protocol for secure exchanging between sides with shared secret. The process for key generation of EC key should look the same as for ECDSA, unless you are talking about shared secret generation, but this shared secret generation should be done during the TLS protocol by the protocol itself.

**Comment by Dvergatal** [comment]
Yeah pkcs11-tool has already support for it, that is why i rather use it than p11tool. And yes you can use openssl command line for tls connection to check if it will use the token properly.

**Comment by ldts** [comment]
I took a stab to proof of concept. I am sure there are issues but still..
https://github.com/ldts/tpm2-pkcs11/commit/f8c2dc07e2273219e92bdc0fda221038bd281b38

**Comment by Dvergatal** [comment]
@ldts thx I can test it later for now I'm little busy but I will certainly do that :smile:
